gadget/device_linux.go

```diff
@@ -138,3 +190,61 @@ func FindMountPointForStructure(ps *PositionedStructure) (string, error) {
 
 	return mountPoint, nil
 }
+
+func isWritableMount(entry *osutil.MountInfoEntry) bool {
+	// example mountinfo entry:
+	// 26 27 8:3 / /writable rw,relatime shared:7 - ext4 /dev/sda3 rw,data=ordered
+	return entry.Root == "/" && entry.MountDir == "/writable" && entry.FsType == "ext4"
+}
+
+func findDeviceForWritable() (device string, err error) {
+	mountInfo, err := osutil.LoadMountInfo(filepath.Join(dirs.GlobalRootDir, osutil.ProcSelfMountInfo))
+	if err != nil {
+		return "", fmt.Errorf("cannot read mount info: %v", err)
+	}
+	for _, entry := range mountInfo {
+		if isWritableMount(entry) {
+			device = entry.MountSource
```

Comment:
Nitpick, this could return device immediately instead of breaking out from the loop.

Reply:
Good catch!